src/RulesProvider/AttributesRulesProvider.php

```diff
@@ -28,38 +30,58 @@ public function __construct(
     }
 
     /**
-     * @return array<RuleInterface[]>
+     * @throws ReflectionException
+     *
+     * @return iterable
      */
-    public function getRules(): array
+    public function getRules(): iterable
     {
         if ($this->rules === null) {
             $this->rules = $this->parseRules();
         }
-        return $this->rules;
+
+        yield from $this->rules;
     }
 
     /**
-     * @return array<RuleInterface[]>
+     * @throws ReflectionException
+     *
+     * @return Generator
```

Comment:
Return types are different in PHPDoc / type hint.

Reply:
Still actual.

Reply:
Done with https://github.com/yiisoft/validator/pull/265/commits/0cf1c8811d0fd089221e11d398d55632802a55f1